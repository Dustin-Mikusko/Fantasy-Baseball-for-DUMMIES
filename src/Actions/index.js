export const addUser = user => ({
  type: 'ADD_USER',
  user
});

export const setUsers = users => ({
  type: 'SET_USERS',
  users
});
