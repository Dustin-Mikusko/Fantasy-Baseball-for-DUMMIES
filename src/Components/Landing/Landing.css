main {
  display: flex;
  height: 100vh;
}

.logo {
  height: 30%
}

.landing-left {
  padding-top: 150px;
  background-color: #041F42;
  color: white;
  width: 150%;
  display: flex;
  flex-direction: column;
  text-align: center;
  font-size: 2.8em;
}

.dummies {
  font-family: Dummy;
  transform: rotate(-15deg);
  margin-top: -10px;
}

.landing-right {
  padding: 100px 50px;
  text-align: justify;
  font-size: 1.5em;
  width: 100%;
}

.headline {
  text-align: center;
  font-size: 2em;
}

.intro {
  font-size: 1.2em;
  margin-bottom: 120px;
}

.buttons {
  display: flex;
  flex-direction: column;
  text-align: center;
}

.sign-in, .sign-up {
  padding-top: 15px;
  margin: auto;
  text-decoration: none;
  color: white;
  height: 50px;
  width: 180px;
  border-radius: 3px;
  font-size: 1.2em;
}

.sign-up {
  background-color: #BF0D3E;
}

.sign-in {
  background-color: #041F42;
  margin-bottom: 20px;
}

.sign-in:hover {
  background-color: rgb(6, 71, 155);
}

.sign-up:hover {
  background-color: rgb(249, 2, 72);
}

.sign-in:active, .sign-up:active {
  transform: scale(.97);
}












