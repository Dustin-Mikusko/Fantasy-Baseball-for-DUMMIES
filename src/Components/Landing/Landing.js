import React, { Component } from 'react';
import './Landing.css';
import logo from '../../images/logo.svg';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { setUsers } from '../../Actions';

export class Landing extends Component  {
  constructor() {
    super();
  }

  componentDidMount() {
    const users = JSON.parse(localStorage.getItem("users"));
    this.props.setUsers(users)
  };

  render() {
    return (
      <main>
        <section className='landing-left'>
          <img src={logo} className='logo' alt='mlb-logo'/>
          <h1>Baseball<br /> for<br /></h1>
          <h1 className='dummies'>DUMMIES</h1>
        </section>
        <section className='landing-right'>
          <h2 className='headline'>New to baseball?</h2>
          <p className='intro'>Step up to the plate today to get a crash course on baseball stats! With our cutting edge technology, we bring you a simple black and white formula to tell you which players will be a HOME RUN in your draft, and those that are sure outs..</p>
          <div className='buttons'>
          <Link to='/signin' className='sign-in'>Sign In</Link>
          <Link to='/signup' className='sign-up'>Sign Up</Link>
          </div>
        </section>
      </main>
    )
  };
}

export const mapDispatchToProps = dispatch => ({
  setUsers: users => dispatch( setUsers(users) )
})

export default connect(null, mapDispatchToProps)(Landing);
