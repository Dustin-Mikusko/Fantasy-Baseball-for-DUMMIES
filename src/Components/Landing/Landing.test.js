import React from 'react';
import { Landing, mapDispatchToProps } from './Landing';
import { shallow } from 'enzyme';
import { setUsers } from '../../Actions';

jest.spyOn(Storage.prototype, 'getItem');

describe('Landing', () => {
  let wrapper, mockSetUsers;

  beforeEach(() => {
    mockSetUsers = jest.fn();
    wrapper = shallow(<Landing 
      setUsers={mockSetUsers}
    />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  // it('should call to localStorage when component mounts', () => {
  //   expect(getItem).toHaveBeenCalledWith('users');
  // });

  describe('mapDispatchToProps', () => {
    it('should call setUsers when component mounts', () => {
      expect(mockSetUsers).toHaveBeenCalled();
    })
  })

})
