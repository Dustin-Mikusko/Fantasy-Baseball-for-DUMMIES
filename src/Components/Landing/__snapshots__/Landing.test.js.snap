// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should match snapshot 1`] = `ShallowWrapper {}`;
